src/oidc.rs

```diff
@@ -36,15 +34,43 @@ impl OIDCConfig {
         };
     }
 
-    pub fn import_config(&mut self) {
-        // futures::executor::block_on(self.get_config());
-        self.certs = "https://keycloak-redhat-sso.apps.dev-eng-ocp4-5.dev.3sca.net/auth/realms/Eloy-wasm-test/protocol/openid-connect/certs".to_string();
-        self.cluster = "keycloak".to_string();
+    fn request(&self, target_url: &str) -> String {
```

Comment:
There is nothing you need about `self` in this fn. How about moving this to the util module? (and since it's a network request, it's better to have it return a `Result`)

Reply:
Just wanted to keep all in the same class to make it clear. This is a workaround ;-)